src/include/pgagroal.h

```diff
@@ -228,6 +228,30 @@ struct prometheus_connection
    atomic_ullong query_count;           /**< The number of queries per connection */
 } __attribute__ ((aligned (64)));
 
+/**
+ * Max size of the cache (in bytes).
+ * If the cache request exceeds this size
+ * the caching should be aborted in some way.
+ */
+#define PROMETHEUS_MAX_CACHE_SIZE 8192
+
+/**
+ * A structure to handle the Prometheus response
+ * so that it is possible to serve the very same
+ * response over and over depending on the cache
+ * settings.
+ *
+ * The `valid_until` field stores the result
+ * of `time(2)` but instead of using a `time_t`
+ * there is an `atomic_ullong` to allow for
+ * concurrency management.
+ */
+struct prometheus_cache
+{
+   char data[PROMETHEUS_MAX_CACHE_SIZE];  /**< the payload */
+   atomic_ullong valid_until;             /**< when the cache will become not valid */
```

Comment:
We will need a `size_t cache_size` and an `atomic_schar prometheus_lock;` as well